### PR TITLE
Jermaine Braumuller

### PR DESCRIPTION
**Pull request by Jaybraum**
Added tests and tried to solve. took me a while to get started.

### CODE REVIEW

File: lib/venue.rb [hunk 1]
```diff
@@ -1,2 +1,28 @@
...
+
+  def capacity
+    @capacity
+  end
+
+  def add_patron(patrons)
+    @patrons << patrons ### <= stuck
+  end
+
+  def patrons.upcase |patrons|
```
Comment (Jaybraum) [inline lib/venue.rb:24]: sorry this is where I got stuck not at the add_patron

File: lib/venue.rb [hunk 2]
```diff
@@ -1,2 +1,28 @@
 class Venue
+  attr_reader :name,
+              :capacity,
+              :patrons
+  def initialize(name, capacity)
+    @name     = name
+    @capacity = capacity
+    @patrons  = []
+    @yell_at_patrons = []
```
Comment (memcmahon) [inline lib/venue.rb:9]: This is not really an attribute of the Venue - it should not be an instance variable.

File: lib/venue.rb [hunk 3]
```diff
@@ -1,2 +1,28 @@
...
+  end
+
+  def name
+    @name
+  end
+
+  def capacity
+    @capacity
+  end
```
Comment (memcmahon) [inline lib/venue.rb:18]: These are actually duplications of the attr_readers for name and capacity above.  Under the hood in ruby, attr_readers actually create these methods!

File: lib/venue.rb [hunk 4]
```diff
@@ -1,2 +1,28 @@
...
+
+  def add_patron(patrons)
+    @patrons << patrons
+  end
+
+  def patrons.upcase |patrons| ###<= stuck here!
+    yell_at_patrons << patrons
+  end
```
Comment (memcmahon) [inline lib/venue.rb:26]: You can't call methods on a method definition (as in `def patrons.upcase`).  This method should be defined with the name `yell_at_patrons` and should iterate over the patrons and transform them into the YELLED version.

```ruby
def yell_at_patrons
  yelled = []
  @patrons.each do |patron|
    yelled << patron.upcase
  end
  yelled
end
```

**Review by memcmahon — COMMENTED** [review]
Good start, but you did not get as far as we would have liked - definitely review the lesson on `each` and I would strongly encourage you to reach out to Madeleine for some one-on-one time.

Score: 1.5

**Pull request closed without merging**